packages/liveblocks-react-ui/src/primitives/Composer/index.tsx

```diff
@@ -1016,11 +1037,12 @@ const ComposerForm = forwardRef<HTMLFormElement, ComposerFormProps>(
     const numberOfAttachments = attachments.length;
     const hasMaxAttachments = numberOfAttachments >= maxAttachments;
     const isDisabled = useMemo(() => {
-      const self = room.getSelf();
-      const canComment = self?.canComment ?? true;
-
-      return isSubmitting || disabled || !canComment;
-    }, [isSubmitting, disabled, room]);
+      // XXX - Discussion: Should we check for scopes here? Scopes are currently part of the room state message
+      // XXX - received after room websocket connection, but maybe we enable the composer submission for
+      // XXX - all users. The request will be rejected by the server, so any optimistic update created
+      // XXX - locally will be reverted!
+      return isSubmitting || disabled === true;
+    }, [isSubmitting, disabled]);
```

Comment:
@marcbouchenoire Because we are no longer under `RoomContext` and this information about whether a user can comment or not is extracted from the room state machine received after WebSocket connection is established, my understanding is that we are not able to disable the composer form if the user in fact does not have scopes to write a comment.

How do you feel about loosening the requirements for the composer form to be disabled? The backend endpoint will reject the update, so comment will be added/removed optimistically but will be immediately reverted.

Reply:
It's not ideal because the delayed removal won't give any feedback about why so it will feel like a bug, but yeah if we can't have this info I don't see another choice 🤔